### PR TITLE
[update]:Provider Wrapper and Protected Routes

### PR DESCRIPTION
**Pull request by ultronjs**
-Added a Provider Wrapper which contains all the provider in one place. So App just need that as a wrapper
-Added RequireAuth for Wishlist and Cart

### CODE REVIEW

**Review by fahd23 — APPROVED** [review]
Code looks clean, good to merge

**Comment by hussiantabha** [comment]
Everything looks very structured, good job. You can merge it.

File: src/pages/RequireAuth.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Navigate } from "react-router"
+import { useAuth } from "../context"
+
+export const RequireAuth = ({children})=>{
+    const {signInStatus} = useAuth()
+    console.log(signInStatus)
```
Comment (t007rushi) [inline src/pages/RequireAuth.js]: left console here

**Review by t007rushi — APPROVED** [review]

File: src/pages/RequireAuth.js (hunk 1 above)
Comment (ultronjs) [reply src/pages/RequireAuth.js]: Removed.Thanks

**Comment by ultronjs** [comment]
Merging after 3 approval

**Pull request merged**